symfony/plugins/orangehrmAdminPlugin/Dao/WorkShiftDao.php

```diff
@@ -101,24 +136,125 @@ public function getWorkShiftEmployeeIdList() {
             }
 
             return $employeeIds;
-
             // @codeCoverageIgnoreStart
         } catch (Exception $e) {
             throw new DaoException($e->getMessage(), $e->getCode(), $e);
         }
         // @codeCoverageIgnoreEnd
     }
 
-    public function saveEmployeeWorkShiftCollection(Doctrine_Collection $empWorkShiftCollection) {
+    public function saveEmployeeWorkShiftCollection(Doctrine_Collection $empWorkShiftCollection)
+    {
         // TODO
         try {
-
             $empWorkShiftCollection->save();
-
             // @codeCoverageIgnoreStart
         } catch (Exception $e) {
             throw new DaoException($e->getMessage(), $e->getCode(), $e);
         }
         // @codeCoverageIgnoreEnd
     }
+
+    /**
+     * @param WorkShift $workShift
+     * @param array $empNumber
+     * @return WorkShift
+     */
+    public function saveWorkShift(WorkShift $workShift, array $empNumber): WorkShift
```

Comment:
wrap this in a transaction.
ref: https://github.com/orangehrm/orangehrm/blob/5.0/symfony/plugins/orangehrmLeavePlugin/Dao/LeaveEntitlementDao.php#L151-L165